### PR TITLE
Fix for ES object shorthand in purs bundle

### LINKED ISSUE #3741: purs bundle misses variable reference in ES6 object shorthand
For example:

```javascript
'use strict';
var foo = "foo";
exports.foo = { foo };
```
will cause the `var foo = "foo"` statement to be removed during bundling, as the bundler is unable to recognise that `foo` is being used in the assigment to `exports.foo`. This is a new issue since updating language-javascript to 0.7, and is thus only reproducible in `master`.

### PR DESCRIPTION
**Pull request by hdgarrood**
Fixes #3741. cc @rhendric, would you mind looking over this?

### CODE REVIEW

File: src/Language/PureScript/Bundle.hs [hunk 1]
```diff
@@ -308,12 +308,21 @@ withDeps (Module modulePath fn es) = Module modulePath fn (map expandDeps es)
       -- ^ exports.foo means there's a dependency on the public member "foo" of
       -- this module.
       = ([(m, nm, Public)], bn)
+    toReference (JSObjectLiteral _ props _) bn
+      = let
+          shorthandNames = mapMaybe unPropertyIdentRef (trailingCommaList props)
+        in
+          (map (\name -> (m, name, Internal)) shorthandNames, bn)
```
Comment (rhendric) [inline src/Language/PureScript/Bundle.hs]: I think this should also be a `mapMaybe` conditional on `name` being an element of `bn`; `foo` should not be exported in the below case:

```js
var foo = "foo";
exports.bar = function (foo) { return { foo }; };
```

Comment (hdgarrood) [reply src/Language/PureScript/Bundle.hs]: Oh yes of course, thanks!

Comment (hdgarrood) [reply src/Language/PureScript/Bundle.hs]: How does it look now?

**Comment by hdgarrood** [comment]
Thanks both!

**Pull request merged**